### PR TITLE
[Y2K-1124] Populate blocking state from the blockMap everywhere

### PR DESCRIPTION
**Pull request by jzila**
cc @keybase/y2ksquad

### CODE REVIEW

File: shared/common-adapters/avatar.tsx [hunk 1]
```diff
@@ -108,13 +107,9 @@ const ConnectedAvatar = Container.connect(
...
-      (
-        (state.tracker2 &&
-          state.tracker2.usernameToDetails &&
-          Tracker2Constants.getDetails(state, ownProps.username || ownProps.teamname || '')) || {
-          blocked: false,
-        }
-      ).blocked || false,
+      state.users &&
+      state.users.blockMap &&
+      state.users.blockMap.get(ownProps.username || ownProps.teamname || '')?.chatBlocked,
```
Comment (hyperobject) [inline shared/common-adapters/avatar.tsx]: `state.users?.blockMap?.get(...)?.chatBlocked ?? false` would be a little more concise here

File: shared/tracker2/remote-proxy.desktop.tsx [hunk 2]
```diff
@@ -18,15 +18,18 @@ const RemoteTracker = (props: {trackerUsername: string}) => {
   const details = Constants.getDetails(state, trackerUsername)
   const {infoMap} = state.users
   const {avatarRefreshCounter, following, followers, httpSrvToken, httpSrvAddress, username} = state.config
-  const {assertions, bio, blocked, followersCount, followingCount, fullname, guiID} = details
+  const {assertions, bio, followersCount, followingCount, fullname, guiID} = details
   const {hidFromFollowers, location, reason, teamShowcase} = details
   const {counts, errors} = state.waiting
+  const {blockMap} = state.users
```
Comment (hyperobject) [inline shared/tracker2/remote-proxy.desktop.tsx]: could combine this with the infoMap declaration, `const {infoMap, blockMap} = state.users`

**Comment by jzila** [comment]
@heronhaye I added a bit more efficiency. PTAL at the last 3 commits.

**Comment by jzila** [comment]
@hyperobject PTAL again, added 3 commits with your PR feedback, and some improvements that make sure the needed data is available, and only that needed data.

**Pull request merged**